Source/Extensions/Blazorise.Splitter/Splitter.razor.cs

```diff
@@ -0,0 +1,184 @@
+    {
+        JSModule ??= new JSSplitModule( JSRuntime, VersionProvider );
+
+        return base.OnInitializedAsync();
+    }
+
+    protected override async Task OnAfterRenderAsync( bool firstRender )
+    {
+        await CreateInstance();
+    }
+
+    /// <inheritdoc />
+    protected override void BuildClasses( ClassBuilder builder )
+    {
+        builder.Append( "split" );
+
+        if ( Direction == SplitDirection.Vertical )
+            builder.Append( "split-vertical" );
+        else
+            builder.Append( "split-horizontal" );
+
+        base.BuildClasses( builder );
+    }
+
+    public void RegisterSection( ElementReference section )
+    {
+        splitSections.Add( section );
+
+        InvokeAsync( StateHasChanged );
+    }
+
+    public void UnregisterSection( ElementReference section )
+    {
+        splitSections.Remove( section );
+
+        InvokeAsync( StateHasChanged );
+    }
+
+    private async Task CreateInstance()
+    {
+        await createInstanceLock.WaitAsync();
+
+        try
+        {
+            if ( JSSplitInstance is not null )
+                await JSSplitInstance.InvokeVoidAsync( "destroy" );
+
+            if ( splitSections.Count > 0 )
+            {
+                var options = new SplitterOptions
+                {
+                    Sizes = Sizes,
+                    MinSize = MinSize,
+                    MaxSize = MaxSize,
+                    ExpandToMin = ExpandToMin,
+                    GutterSize = GutterSize,
+                    GutterAlign = GutterAlign,
+                    SnapOffset = SnapOffset,
+                    DragInterval = DragInterval,
+                    Direction = Direction,
+                    Cursor = Cursor,
+                };
+
+                JSSplitInstance = await JSModule.InitializeSplit( splitSections, options );
+            }
+        }
+        finally
+        {
+            createInstanceLock.Release();
+        }
+    }
+
+    #endregion
+
+    #region Properties
+
+    /// <summary>
+    /// Gets or sets the <see cref="JSSplitModule"/> instance.
+    /// </summary>
+    protected JSSplitModule JSModule { get; private set; }
+
+    /// <summary>
+    /// Gets or sets the reference to the JS split instance
+    /// </summary>
+    protected IJSObjectReference JSSplitInstance { get; private set; }
+
+    /// <summary>
+    /// Gets or sets the JS runtime.
+    /// </summary>
+    [Inject] private IJSRuntime JSRuntime { get; set; }
+
+    /// <summary>
+    /// Gets or sets the version provider.
+    /// </summary>
+    [Inject] private IVersionProvider VersionProvider { get; set; }
+
+    /// <summary>
+    /// Initial sizes of each element in percents or CSS values.
+    /// </summary>
+    [Parameter] public IEnumerable<JavascriptNumber> Sizes { get; init; }
+
+    /// <summary>
+    /// Minimum size of each element.
+    /// </summary>
+    [Parameter] public JavascriptNumberOrArray MinSize { get; init; }
```

Comment:
I would prefer users to have simple apis they are used to, then these newer ones, because it seems to me we might not need them.
If we really need them, it would be great if they could only be used internally. That is, if we do need them.